### DISCUSSION TITLE
xDrip variant: cannot import settings by file

### DISCUSSION THREAD

**Discussion by robertrub** [NightscoutFoundation/xDrip#2861] [Q&A]
Strange behaviour. I installed xDrip _variant1_ on nightly build and I have the latest nightly as main xDrip. 

I exported settings from _main_. I wanted to import those settings into _variant_ but _variant_ kept saying that the file or permissions were missing. The file is in the correct place and the permissions are ok.

I thought maybe a security problem. But, I'm stubborn... I screen captured the settings' QR code from _main_ and "auto configuration" / scanned the code in _variant_ and it worked.

No idea why _variant1_ refused to import settings by file but it did by QR code.

**Comment by Lupum1001** [comment]
can variant1 export a file, and then re-import its own file? The error message in the code appears to be generic (i.e. if anything failed while importing, display that same error).

The file names are very nearly hard-coded (I think they vary based on package name), so my first guess is that the file names could just be incompatible.

**Reply by robertrub** [reply]
![Screenshot_20230529_212641_File Manager +.jpg](https://github.com/NightscoutFoundation/xDrip/assets/76626474/7ac3d4fa-ac3e-49e7-8e9d-6d16dbb96a26)

Yes, you're totally correct. The export file name of the _variant_ is different than the _main_. Thus, the "can't import" error message. 

The import via the QR code worked as it probably didn't have the version info in it.

Mystery solved :) Thanks.

**Reply by Lupum1001** [reply]
Glad I could help!
A file-picker would improve flexibility of course, but I also notice that file name appears to be the only safety check at the moment. I assume they'd need to add validation logic on the contents of the file if they wanted a file picker. 🤔
